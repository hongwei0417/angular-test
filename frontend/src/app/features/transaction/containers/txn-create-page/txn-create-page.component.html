<!-- <h3 class="text-center">Txn Create Page</h3>

<div class="d-flex justify-content-center">
  <button [routerLink]="['/transaction','list']">Transaction list</button>
  <button [routerLink]="['/transaction','create']">Create Transaction</button>
</div> -->

<app-txn-create
  [errorMessage]="(error$ | async)!"
  (submitted)="onSubmit($event)"
  (clearState)="onClearState()"
></app-txn-create>
