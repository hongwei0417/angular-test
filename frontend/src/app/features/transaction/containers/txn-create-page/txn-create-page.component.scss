@import '/src/assets/styles/variables.scss';

.title{
  border-bottom: 1px solid #000;
  padding-bottom: 10px;
}

h3{
  display: flex;
  align-items: center;
  text-align: center;
  color: $primary-medium;
  margin-bottom: 10px;
}

.jobSetting h3{
  margin-top: 30px;
}

h3:before{
  content: '1';
  margin-right: 5px;
  width: 25px;
  height: 25px;
  display: block;
  border-radius: 50%;
  background: $primary-medium;
  color: $gray-100;
  line-height: 25px;
  font-size: 16px;
}

.jobSetting h3:before{
  content: '2';
}

h4{
  margin-bottom: 5px;
}

form:last-child{
  margin-bottom: 0;
}

input{
  background: $gray-100;
}

.bg{
  padding: 15px;
  background: $gray-200;
}

.btnGroup{
  display: flex;
}

.btnGroup button:last-child{
  margin-left: 15px;
}


.jobSetting .singleJobSetting{
  margin-top: 30px;
}
