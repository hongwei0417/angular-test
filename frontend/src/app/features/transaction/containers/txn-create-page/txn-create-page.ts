import { Observable, of } from 'rxjs';
import { Store, select } from '@ngrx/store';
import { Component, OnDestroy, OnInit } from '@angular/core';
import { TxnCreatePageActions } from '../../actions';
import * as fromTxn from '../../reducers';

@Component({
  selector: 'app-txn-create-page',
  templateUrl: './txn-create-page.component.html',
  styleUrls: ['./txn-create-page.component.scss'],
})
export class TxnCreatePageComponent implements OnInit, OnDestroy {
  error$!: Observable<string>;

  constructor(private store$: Store<fromTxn.State>) {}

  ngOnInit(): void {
    this.error$ = this.store$.pipe(select(fromTxn.getTxnCreateError));
  }

  ngOnDestroy(): void {
    this.store$.dispatch(TxnCreatePageActions.clearTxnCreatePageState());
  }

  onSubmit({ title, content, executeCount }: any): void {
    this.store$.dispatch(
      TxnCreatePageActions.createTxn({ title, content, executeCount })
    );
  }
}
