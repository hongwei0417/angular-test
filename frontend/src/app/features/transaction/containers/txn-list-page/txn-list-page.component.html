<h3 class="text-center">Transaction List Page</h3>

<div class="d-flex justify-content-center">
  <button [routerLink]="['/transaction', 'list']">Transaction list</button>
  <button [routerLink]="['/transaction', 'create']">Create Transaction</button>
</div>

<div class="d-flex justify-content-center">
  <button id="generateDataBtn" (click)="loadTxnData()">Generate Data</button>
</div>

<app-txn-list-table
  [tableData]="[]"
  (removeEvent)="onRemoveTxn($event)"
></app-txn-list-table>
