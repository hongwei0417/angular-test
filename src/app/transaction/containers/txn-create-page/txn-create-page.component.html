<h3 class="text-center">Txn Create Page</h3>

<div class="d-flex justify-content-center">
  <button routerLink="/txn/feature1/list">Transaction list</button>
  <button routerLink="/txn/feature1/create">Create Transaction</button>
</div>

<app-txn-create
  [errorMessage]="(error$ | async)!"
  (submitted)="onSubmit($event)"
  (clearState)="onClearState()"
></app-txn-create>
