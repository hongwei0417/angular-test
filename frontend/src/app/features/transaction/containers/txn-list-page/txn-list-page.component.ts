import { FormGroupState } from 'ngrx-forms';
import { FilterFormValue } from './../../reducers/txn-list-page.reducer';
import { Observable, of } from 'rxjs';
import { Component, OnInit, OnDestroy } from '@angular/core';
import { select, Store } from '@ngrx/store';
import * as fromTxn from '../../reducers';
import { TxnListPageActions } from '../../actions';
import { TxnListTableRow } from '../../models/TxnListTable';
@Component({
  selector: 'app-txn-list-page',
  templateUrl: './txn-list-page.component.html',
  styleUrls: ['./txn-list-page.component.scss'],
})
export class TxnListPageComponent implements OnInit, OnDestroy {
  txnData$!: Observable<TxnListTableRow[]>;
  filterForm$!: Observable<FormGroupState<FilterFormValue>>;

  constructor(private store$: Store<fromTxn.State>) {}

  ngOnInit(): void {
    this.txnData$ = this.store$.pipe(select(fromTxn.getAllTxns));
    this.filterForm$ = this.store$.pipe(select(fromTxn.getFilterForm));
    this.store$.dispatch(TxnListPageActions.loadTxnList());
  }

  ngOnDestroy(): void {
    this.store$.dispatch(TxnListPageActions.clearTxnListPageState());
  }

  createNewTxn(): void {
    this.store$.dispatch(TxnListPageActions.createNewTxn());
  }

  onViewTxnInfo(id: string): void {
    this.store$.dispatch(TxnListPageActions.viewTxnInfo({ id }));
  }

  onEditTxnInfo(id: string): void {
    this.store$.dispatch(TxnListPageActions.editTxnInfo({ id }));
  }

  onCopyInfo(id: string): void {
    this.store$.dispatch(TxnListPageActions.copyTxnInfo({ id }));
  }

  onExecuteTxn(id: string): void {
    this.store$.dispatch(TxnListPageActions.executeTxn({ id }));
  }

  onReloadTxn(id: string): void {
    this.store$.dispatch(TxnListPageActions.reloadTxn({ id }));
  }
}
