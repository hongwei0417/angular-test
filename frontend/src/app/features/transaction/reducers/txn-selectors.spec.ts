import * as fromTxn from './index';
import * as fromTxnListPage from './txn-list-page.reducer';
import * as fromTxnCreatePage from './txn-create-page.reducer';
import { fakeData } from '../../../core/services/fake-data.service';
import { autoschtranms } from './../../../shared/testing/data/txnListTableData';

describe('Txn Selectors', () => {
  let initialState: fromTxn.TxnState;

  beforeAll(() => {
    initialState = {
      [fromTxnCreatePage.FeatureKey]: {
        loading: true,
        error: 'this is a error',
      },
      [fromTxnListPage.FeatureKey]: {
        ...getEntityState(),
        loaded: true,
        loading: true,
        latestId: '1',
      },
    };
  });

  describe('txn list page selectors', () => {
    it('should get list page state', () => {
      const result =
        fromTxn.getTxnListPageEntitiesState.projector(initialState);
      expect(result).toMatchSnapshot();
      expect(result).toEqual(initialState[fromTxnListPage.FeatureKey]);
    });

    it('should get latest txn id', () => {
      const state: fromTxnListPage.State = {
        ...initialState[fromTxnListPage.FeatureKey],
        latestId: '666',
      };
      const result = fromTxn.getLatestTxnId.projector(state);
      expect(result).toBe(state.latestId);
    });

    it('should get latest txn data', () => {
      const state: fromTxnListPage.State = {
        ...initialState[fromTxnListPage.FeatureKey],
        latestId: initialState[fromTxnListPage.FeatureKey].ids[0] as string,
      };
      const result = fromTxn.getLatestTxn.projector(
        state.entities,
        state.latestId
      );
      expect(result).toEqual(state.entities[state.latestId as string]);
    });
  });

  describe('create txn page selectors', () => {
    it('should get create page state', () => {
      const result = fromTxn.getTxnCreatePageState.projector(initialState);
      expect(result).toMatchSnapshot();
      expect(result).toEqual(initialState[fromTxnCreatePage.FeatureKey]);
    });

    it('should get create error msg', () => {
      const result = fromTxn.getTxnCreateError.projector(
        initialState[fromTxnCreatePage.FeatureKey]
      );
      expect(result).toBe(initialState[fromTxnCreatePage.FeatureKey].error);
    });
  });

  const getEntityState = () => {
    const ids = autoschtranms.map((i) => i.TRANSACTIONID);
    const entities = {} as any;
    ids.forEach((i, n) => (entities[i] = autoschtranms[n]));
    const state: fromTxnListPage.State = {
      ...fromTxnListPage.initialState,
      ids,
      entities,
    };
    return state;
  };
});
