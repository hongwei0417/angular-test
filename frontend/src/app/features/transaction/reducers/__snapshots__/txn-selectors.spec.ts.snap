// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Txn Selectors create txn page selectors should get create page state 1`] = `
Object {
  "error": "this is a error",
  "loading": true,
}
`;

exports[`Txn Selectors txn list page selectors should get list page state 1`] = `
Object {
  "entities": Object {
    "1": Object {
      "content": "test case 1",
      "createTime": 2021-08-23T16:00:00.000Z,
      "executeCount": 10,
      "id": "1",
      "title": "Kevin",
    },
    "2": Object {
      "content": "do something tonight...",
      "createTime": 2021-08-24T16:00:00.000Z,
      "executeCount": 4,
      "id": "2",
      "title": "Jack",
    },
    "3": Object {
      "content": "send email to everybody",
      "createTime": 2021-08-25T16:00:00.000Z,
      "executeCount": 1,
      "id": "3",
      "title": "Eric",
    },
  },
  "ids": Array [
    "1",
    "2",
    "3",
  ],
  "latestId": "1",
  "loaded": true,
  "loading": true,
}
`;
